src/Lykke.Job.CandlesProducer.Services/Trades/Spot/SpotTradesSubscriber.cs

```diff
@@ -35,123 +34,88 @@ public class SpotTradesSubscriber : ITradesSubscriber
 
         public void Start()
         {
-            _marketTradesSubscriber = _subscribersFactory.Create<MarketTradesMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", _tradesSubscriptionSettings.EndpointName, ProcessMarketTradesAsync);
-            _limitTradesSubscriber = _subscribersFactory.Create<LimitTradesMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", "limitorders.clients", ProcessLimitTradesAsync);
+            _limitTradesSubscriber = _subscribersFactory.Create<LimitOrdersMessage>(_tradesSubscriptionSettings.ConnectionString, "lykke", "limitorders.clients", ProcessLimitTradesAsync);
         }
 
-        private async Task ProcessMarketTradesAsync(MarketTradesMessage message)
+        private async Task ProcessLimitTradesAsync(LimitOrdersMessage message)
         {
-            if (message.Trades == null || !message.Trades.Any())
+            if (message.Orders == null || !message.Orders.Any())
             {
                 return;
             }
 
-            var trades = message
-                .Trades
-                .Select(t =>
+            var limitOrderIds = message.Orders
+                .Select(o => o.Order.Id)
+                .ToHashSet();
+
+            foreach (var orderMessage in message.Orders)
+            {
+                if (orderMessage.Trades == null)
                 {
-                    TradeType tradeType;
-                    // Volumes in the asset pair base and quoting assets
-                    double baseVolume;
-                    double quotingVolume;
+                    continue;
+                }
 
-                    var assetPair = _assetPairsManager.TryGetEnabledPairAsync(message.Order.AssetPairId)
+                foreach (var tradeMessage in orderMessage.Trades)
+                {
+                    var assetPair = _assetPairsManager.TryGetEnabledPairAsync(orderMessage.Order.AssetPairId)
```

Comment:
This is async method. Why TryGetEnabledPairAsync is called sync?

Reply:
It's actually callde with .GetAwaiter().GetResult() - please, take a look ath the next 2 lines.

Reply:
I think @MikhailGerasimov meant why not ```await _assetPairsManager.TryGetEnabledPairAsync()```

Reply:
And looks like this call can be moved outside of the inner loop

Reply:
Yep